ui/src/app/pages/workspace/workspace-edit.tsx

```diff
@@ -395,50 +395,70 @@ export const WorkspaceEdit = fp.flow(withRouteConfigData(), withCurrentWorkspace
+              }});
+            }
+          }
         }
-      }
 
-      const cdrResp = this.props.cdrVersionListResponse;
-      const liveCdrVersions = cdrResp.items.filter(cdr => cdr.archivalStatus === ArchivalStatus.LIVE);
-      if (liveCdrVersions.length === 0) {
-        throw Error('no live CDR versions were found');
-      }
-      if (this.isMode(WorkspaceEditMode.Edit)) {
-        // In edit mode, you cannot modify the CDR version, therefore it's fine
-        // to show archived CDRs in the drop-down so that it accurately displays
-        // the current value.
-        this.setState({cdrVersionItems: cdrResp.items.slice()});
-      } else {
-        // In create/clone, disallow selection of archived versions by omitting
-        // them. The server will also reject archived CDRs.
-        this.setState({cdrVersionItems: liveCdrVersions});
-      }
+        const cdrResp = this.props.cdrVersionListResponse;
+        const liveCdrVersions = cdrResp.items.filter(cdr => cdr.archivalStatus === ArchivalStatus.LIVE);
+        if (liveCdrVersions.length === 0) {
+          throw Error('no live CDR versions were found');
+        }
+        if (this.isMode(WorkspaceEditMode.Edit)) {
+          // In edit mode, you cannot modify the CDR version, therefore it's fine
+          // to show archived CDRs in the drop-down so that it accurately displays
+          // the current value.
+          this.setState({cdrVersionItems: cdrResp.items.slice()});
+        } else {
+          // In create/clone, disallow selection of archived versions by omitting
+          // them. The server will also reject archived CDRs.
+          this.setState({cdrVersionItems: liveCdrVersions});
+        }
+
+        const selectedCdrIsLive = liveCdrVersions.some(cdr => cdr.cdrVersionId === this.state.workspace.cdrVersionId);
+        if (this.isMode(WorkspaceEditMode.Create) || (
+          this.isMode(WorkspaceEditMode.Duplicate) && !selectedCdrIsLive)) {
+          // We preselect the default CDR version when a new workspace is being
+          // created (via create or duplicate) with one exception: cloning a
+          // workspace which references a non-default, non-archived CDR version.
+          this.setState(fp.set(['workspace', 'cdrVersionId'], cdrResp.defaultCdrVersionId));
+        }
+      });
+    }
 
-      const selectedCdrIsLive = liveCdrVersions.some(cdr => cdr.cdrVersionId === this.state.workspace.cdrVersionId);
-      if (this.isMode(WorkspaceEditMode.Create) || (
-        this.isMode(WorkspaceEditMode.Duplicate) && !selectedCdrIsLive)) {
-        // We preselect the default CDR version when a new workspace is being
-        // created (via create or duplicate) with one exception: cloning a
-        // workspace which references a non-default, non-archived CDR version.
-        this.setState(fp.set(['workspace', 'cdrVersionId'], cdrResp.defaultCdrVersionId));
+    async maybeUpdateRecentWorkspaces() {
```

Comment:
Another place on this page where we may want to updateRecentWorkspaces is when you clone a workspace, we probably want to add the clone onto the set of recentWorkspaces. Is that handled by the API?